pdart/pipeline/SuffixInfo.py

```diff
@@ -0,0 +1,420 @@
+        "Combined, calibrated {I} image files from repeated exposures, without CTE correction, in HST Program {P}",
+        ["WFC3"],
+    ),
+    ("WFC3", "UVIS", "drz"): (  # because drc is also available
+        True,
+        True,
+        False,
+        "Derived",
+        "Data",
+        "Calibrated {I} image file {F}, corrected for geometric distortion but not CTE, from HST Program {P}",
+        "Calibrated {I} image files, corrected for geometric distortion but not CTE, from HST Program {P}",
+        ["WFC3"],
+    ),
+    ("WFC3", "IR", "raw"): (  # Why do this?  Possibly remove.
+        True,
+        True,
+        False,
+        "Raw",
+        "Data",
+        "Raw, uncalibrated {I} image file {F}, without CR rejection, from HST Program {P}",
+        "Raw, uncalibrated {I} image files, without CR rejection, from HST Program {P}",
+        ["WFC3"],
+    ),
+    ("WFC3", "IR", "ima"): (  # Doesn't match handbook.
+        False,
+        False,
+        False,
+        "Partially Processed",
+        "Data",
+        "Calibrated {I} image file {F}, without CR rejection, from HST Program {P}",
+        "Calibrated {I} image files, without CR rejection, from HST Program {P}",
+        ["WFC3"],
+    ),
+    ("WFC3", "IR", "flt"): (  # Why do this?  Possibly remove.
+        True,
+        True,
+        False,
+        "Calibrated",
+        "Data",
+        "Calibrated {I} image file {F}, without CR rejection, from HST Program {P}",
+        "Calibrated {I} image files, without CR rejection, from HST Program {P}",
+        ["WFC3"],
+    ),
+    ("NICMOS", "cal"): (
+        True,
+        False,
+        False,
+        "Calibrated",
+        "Data",
+        "Calibrated {I} image file {F} from HST Program {P}",
+        "Calibrated {I} image files from HST Program {P}",
+        ["NICMOS"],
+    ),
+    ("NICMOS", "ima"): (
+        False,
+        False,
+        False,
+        "Partially Processed",
+        "Data",
+        "Calibrated but uncombined {I} image file {F} for MULTIACCUM observations in HST Program {P}",
+        "Calibrated but uncombined {I} image file for MULTIACCUM observations in HST Program {P}",
+        ["NICMOS"],
+    ),
+    ("NICMOS", "mos"): (
+        True,
+        False,
+        False,
+        "Derived",
+        "Data",
+        "Combined, calibrated {I} image file {F} for dithered observations in HST Program {P}",
+        "Combined, calibrated {I} image files for dithered observations in HST Program {P}",
+        ["NICMOS"],
+    ),
+}
+
+# First letter of filenames and the corresponding instrument names
+INSTRUMENTS_INFO = {
+    "i": "wfc3",
+    "j": "acs",
+    "l": "cos",
+    "n": "nicmos",
+    "o": "stis",
+    "u": "wfpc2",
+    "w": "wfpc",
+    "x": "foc",
+    "y": "fos",
+    "z": "ghrs",
+}
+
+# First letter of filenames
+ACCEPTED_INSTRUMENTS = "".join(INSTRUMENTS_INFO.keys()).upper()
```

Comment:
(1) We need to decide on a standard for how to refer to file names in the code--upper case or lower case. The letter codes are lower in the dictionary but upper in ACCEPTED_INSTRUMENTS. This is already a bit confusing. This standard would also determine whether we refer to suffixes in upper or lower case. I recommend lower case, because that is how we will store the files locally. (2) For the same reason as my comment above about changing INSTRUMENTS_INFO to INSTRUMENT_FROM_LETTER_CODE, a better name for "ACCEPTED_INSTRUMENTS" would be "ACCEPTED_LETTER_CODES". Elsewhere in the code, we should consistently use "inst" or "instrument" to refer to the instrument ID, and "letter" or "letter code" should refer to these single-letter codes. (3) I think you should check how this variable is used elsewhere in the code, because right now it contains every letter code, implying that we accept every instrument--which we don't! If you do need a way to derive the complete list of instruments we accept, it should be derived from SUFFIX_INFO instead.

Reply:
- Done. Now instrument ids are upper case, and file names/suffixes are lower case. 
- ACCEPTED_INSTRUMENTS is renamed to ACCEPTED_LETTER_CODES.
- Create ```_get_accepted_inst_li``` to get the accepted instruments from SUFFIX_INFO, and ```_get_accepted_letter_codes``` to get the accepted letter list based on the accepted instruments